### PR TITLE
[flink] Introduce Query Service

### PR DESCRIPTION
**Pull request by JingsongLi**
### Purpose

Final step for: https://cwiki.apache.org/confluence/display/PAIMON/PIP-10%3A+Introduce+Paimon+QueryService

Build DataStream topology for query service.

### CODE REVIEW

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/QueryServiceActionFactory.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+import org.apache.flink.api.java.tuple.Tuple3;
+
+import java.util.Map;
+import java.util.Optional;
+
+/** Factory to create QueryService Action. */
+public class QueryServiceActionFactory implements ActionFactory {
+
+    public static final String IDENTIFIER = "query-service";
```
Comment (yuzelin) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/QueryServiceActionFactory.java:31]: `query_service`

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/service/QueryExecutorOperator.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+                        1,
+                        1,
+                        query,
+                        new DisabledServiceRequestStats());
+
+        try {
+            server.start();
+        } catch (Throwable e) {
+            throw new RuntimeException(e);
+        }
```
Comment (yuzelin) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/service/QueryExecutorOperator.java:93]: Why catch? `initializeState` will throw Exception.

Comment (JingsongLi) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/service/QueryExecutorOperator.java:93]: It is Throwable, not Exception

**Review by yuzelin — APPROVED** [review]
+1

**Pull request merged**